src/API/Models/DesignerModel.ts

```diff
@@ -82,28 +105,39 @@ export const readAllDesigners = async (filters: DesignerFilters) => {
       },
       baseAccount: {
         select: {
-          avatarUrl: true
+          avatarUrl: true,
+          gender: true,
+          firstName: true,
+          lastName: true
         }
       }
     }
   });
 
-
-  return designers.map(designer => {
+  const filteredDesigners = designers.map(designer => {
     const workingDays: WorkingHours = JSON.parse(designer.workingDays as unknown as string);
     const { open, openUntil } = isOpenNow(workingDays);
 
     return {
       baseAccountId: designer.baseAccountId,
       ordersFinished: designer.ordersFinished,
-      location: designer.location, // Include location instead of address
+      location: designer.location,
       yearsExperience: designer.yearsExperience,
       rating: designer.reviews.length ? designer.reviews.reduce((sum: number, review: { rating: number }) => sum + review.rating, 0) / designer.reviews.length : 0,
       avatarUrl: designer.baseAccount.avatarUrl,
+      gender: designer.baseAccount.gender,
+      name: `${designer.baseAccount.firstName} ${designer.baseAccount.lastName}`,
       openNow: open,
       openUntil: open ? openUntil : null
     };
   });
+
+  const openFilteredDesigners = openNow !== undefined
+    ? filteredDesigners.filter(designer => designer.openNow === openNow)
+    : filteredDesigners;
+
+  const paginatedDesigners = openFilteredDesigners.slice((page - 1) * limit, page * limit);
+  return paginatedDesigners.length ? paginatedDesigners : null;
```

Comment:
You might also wanna return some pagination details
current_page: number
total_pages: number
entry_counts: number
next_page: number | null
prev_page: number | null

Reply:
these details should be sent back to the user to make it easier for them to queue and have more inforamtion

Reply:
@Jupi2051  on it